### PR TITLE
ui-manchette: add fit button

### LINKED ISSUE #410: manchette: add fit button
closes https://github.com/osrd-project/osrd-confidential/issues/636

## Description

The header represents the spatial axis of the Space-Time Diagram (GET). From the header, there should be a FIT button that allows switching to a view where the entire path is displayed.
On the first render, the selected train must be optimally displayed horizontally.

## Acceptance Criteria

- [ ] The user has a button to reset the zoom level in order to see a train in its entirety.
- [ ] On the first render, the user can see the selected train in full.
- [ ] On the first render, the horizontal scale must be adjusted to the trains in the timetable.

### osrd-ui (size 0.75)
- In the header, add a button to dispatch an event that resets the scale.
- In the Header+GET component, add logic to:
  - Adjust the maximum allowed zoom level on the GET.
  - Modify the vertical scales (space) when clicking the button.
  - Modify the horizontal scale (time) for the first render.

### osrd (size 0.25)

- Integrate the new version of the header, providing the new props if necessary.

### PR DESCRIPTION
**Pull request by clarani**
closes #410 

image

### CODE REVIEW

File: ui-manchette/src/components/Manchette.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ type ManchetteProps = {
   activeOperationalPointId?: string;
   zoomYIn: () => void;
   zoomYOut: () => void;
+  resetZoom: () => void;
```
Comment (emersion) [inline ui-manchette/src/components/Manchette.tsx:15]: Because this is not optional, and this prop is not supplied until the next commit, the build breaks when checking out this commit. Suggestion: (1) merge the two commits or (2) add the button without the prop in this commit, add the prop in the next commit instead.

File: ui-manchette/src/styles/manchette.css [hunk 2]
```diff
@@ -9,20 +15,34 @@
...
+
+      .zoom-reset {
+        color: #797671;
+        font-weight: 600;
+        font-size: 14px;
+      }
     }
 
-    .toggle-mode {
+    .toggle-mode{
```
Comment (emersion) [inline ui-manchette/src/styles/manchette.css]: Style: missing space before `{`

File: ui-manchette/src/styles/manchette.css [hunk 3]
```diff
@@ -9,20 +15,34 @@
...
+
+      button {
+        border-right: solid 1px #d3d1cf;
+        border-top: solid 1px white;
+        height: 100%;
+        padding-inline: 12px;
+      }
+
+      button:disabled {
+        span {
```
Comment (emersion) [inline ui-manchette/src/styles/manchette.css]: I'd prefer not to use `span` here, because this adds assumptions about the internal HTML structure of icon components. I believe the CSS rule should work fine with that selector dropped?

File: ui-manchette/src/components/Manchette.tsx [hunk 4]
```diff
@@ -40,24 +42,17 @@ const Manchette = ({
...
-        <button
-          className="h-full px-3 w-full zoom-in"
-          disabled={yZoom >= MAX_ZOOM_Y}
-          onClick={zoomYIn}
-        >
+        <button className="zoom-in" onClick={zoomYIn} disabled={yZoom >= MAX_ZOOM_Y}>
           <ZoomIn />
         </button>
+        <button className="zoom-reset" onClick={resetZoom}>
+          Fit
```
Comment (emersion) [inline ui-manchette/src/components/Manchette.tsx:54]: Seems kind of weird to have this untranslated string next to icon buttons. But it seems like it's what the mockup wants so shrug.

File: ui-manchette/src/styles/manchette.css [hunk 5]
```diff
@@ -9,20 +15,34 @@
...
+          color: #B6B2AF
+        }
+      }
+
+      .zoom-reset {
+        color: #797671;
```
Comment (emersion) [inline ui-manchette/src/styles/manchette.css]: Could we use the design system's colors instead of hardcoded hex codes? Something like `@apply text-grey-50`? (Ditto for other colors)

File: ui-manchette/src/styles/manchette.css [hunk 6]
```diff
@@ -1,6 +1,12 @@
 .manchette-container {
   .manchette-actions {
+    align-items: center;
     background-color: rgba(250, 249, 245, 0.6);
+    border-top: solid 1px;
+    border-right: solid 1px;
+    border-radius: 0px 0px 0px 10px;
```
Comment (emersion) [inline ui-manchette/src/styles/manchette.css]: Nit: no need to specify the unit for zero lengths.

File: ui-manchette-with-spacetimechart/src/stories/base.stories.tsx [hunk 7]
```diff
@@ -69,10 +69,12 @@ const ManchetteWithSpaceTimeWrapper = ({
   );
 };
 
-export default {
+const meta = {
```
Comment (emersion) [inline ui-manchette-with-spacetimechart/src/stories/base.stories.tsx]: Seems like this unrelated change slipped into this commit?

File: ui-manchette/src/styles/manchette.css [hunk 8]
```diff
@@ -9,17 +15,29 @@
...
 
-    .zoom-in:disabled {
-      opacity: 0.4;
+    .zoom-buttons {
+      display: flex;
+      align-items: center;
+      height: 100%;
+
+      button {
+        border-right: solid 1px #d3d1cf;
```
Comment (emersion) [inline ui-manchette/src/styles/manchette.css]: Could we use `theme()` to avoid hardcoding a color here?

https://tailwindcss.com/docs/functions-and-directives#theme

File: ui-manchette/src/stories/Manchette.stories.tsx [hunk 9]
```diff
@@ -23,78 +19,14 @@ const meta: Meta<typeof Manchette> = {
...
-    zoomYIn: {
-      action: 'zoomYIn',
-    },
-    zoomYOut: {
-      action: 'zoomYOut',
-    },
   },
 };
 
-const ManchetteWithWaypointMenu = () => {
```
Comment (SharglutDev) [inline ui-manchette/src/stories/Manchette.stories.tsx]: You don't want to keep this story ? :face_holding_back_tears:

Comment (clarani) [reply ui-manchette/src/stories/Manchette.stories.tsx]: It is not used for the moment, so currently it's dead code :)
I tried to see what how it works, but it was a bit broken, so I decided to remove it. If you prefer, we can try to fix it, or we can remove it in this commit and put it again later, once we have a more precise use case ?

File: ui-manchette/src/styles/manchette.css (hunk 8 above)
Comment (clarani) [reply ui-manchette/src/styles/manchette.css]: Sure

File: ui-manchette/src/stories/Manchette.stories.tsx (hunk 9 above)
Comment (clarani) [reply ui-manchette/src/stories/Manchette.stories.tsx]: Sorry, I misunderstood ! The story is back :white_check_mark:

**Comment by emersion** [comment]
Seems like the first commit "ui-manchette: clean manchette's story" is just shuffling stuff around now? Maybe it should be dropped?

**Comment by clarani** [comment]
You're right, it's done :)

**Review by emersion — APPROVED** [review]
LGTM, thanks!
